src/library_fs.js

```diff
@@ -838,7 +838,7 @@ mergeInto(LibraryManager.library, {
       if (!link.node_ops.readlink) {
         throw new FS.ErrnoError(ERRNO_CODES.EINVAL);
       }
-      return link.node_ops.readlink(link);
+      return PATH.resolve('.', link.node_ops.readlink(link));
```

Comment:
Maybe I don't understand this part. What did we have here before (in the node and non-node cases) and what do we have after he change, that makes this `resolve()` necessary?